application/src/main/java/com/sanction/lightning/resources/FacebookResource.java

```diff
@@ -112,6 +117,60 @@ public Response getPhotos(@Auth Key key, @QueryParam("username") String username
   }
 
   /**
+   * Publishes to a users facebook.
+   * @param key The authentication key for the requesting application.
+   * @return Response that the photo either uploaded or failed.
+   */
+  @POST
+  @Path("/publish")
+  @Consumes(MediaType.MULTIPART_FORM_DATA)
+  public Response publish(@Auth Key key, @QueryParam("username") String username,
+                               @FormDataParam("file") final InputStream inputStream,
+                               @FormDataParam("file") final FormDataContentDisposition
+                                         contentDispositionHeader,
+                               @FormDataParam("message") final String message) {
```

Comment:
I don't think we need to make these variables `final`, we haven't done it for any other function.